UnitySetup/UnitySetup.psm1

```diff
@@ -2729,7 +2738,7 @@ function Update-UnityPackageManagerConfig {
     $tomlFileObjects = Import-TOMLFile -tomlFilePaths $tomlFilePaths -Force
 
     if ($PSCmdlet.ShouldProcess("Synchronizing UPM configuration")) {
-        $upmConfigs = Update-PackageAuthConfig -ScopedRegistryURLs $scopedRegistryURLs -TomlfileObjects $tomlFileObjects -AutoClean:$AutoClean.IsPresent -VerifyOnly:$VerifyOnly.IsPresent -ManualPAT:$ManualPAT.IsPresent -PATLifetime $PATLifetime -DefaultScope $defaultScope -AzAPIVersion $azAPIVersion -ScopedURLRegEx $scopedURLRegEx -UPMRegEx $upmRegEx
+        $upmConfigs = Update-PackageAuthConfig -ScopedRegistryURLs $scopedRegistryURLs -TomlfileObjects $tomlFileObjects -AutoClean:$AutoClean.IsPresent -VerifyOnly:$VerifyOnly.IsPresent -ManualPAT:$ManualPAT.IsPresent -PATLifetime $PATLifetime -DefaultScope $defaultScope -AzAPIVersion $azAPIVersion -ScopedURLRegEx $scopedURLRegEx -UPMRegEx $upmRegEx -defaultSubscriptionId $defaultSubscriptionId
```

Comment:
So even if the subscript id is null we'll pass it here and that'll make the above test `$PSBoundParameters.ContainsKey` be true since you did pass something. Maybe above you just test if the string is not null or empty? Would be good to mark up the attribute here too with `ValidateNotNullOrEmpty` so that people can't pass invalid subscription ids.

Reply:
Sure - added some validation